### PR TITLE
NAS-132657 / 25.04 / Remove explicit calls to syslog.syslog module

### PR DESCRIPTION
**Pull request by yocalebo**
As pointed out by a community user, we're doing a bunch of random/weird things with syslog messages. While the community member focused on the mail functionality, I found we had sprinkled in random syslog calls all over the place.

There are a bunch of problems with what we were doing:
1. we were violating the syslog specification by sending newlines in the syslog messages (they should be single lines only)
2. `find_alias_*` scripts are not used on SCALE and can be safely ignored (this improves our security posture)
3. all of the messages in middlewared.log are monitored by `syslog-ng`. If a remote syslog server is configured, then syslog-ng automatically monitors that file and sends those messages to the remote system

Because we already forward relevant log messages via `syslog-ng`, this removes the explicit calls to `syslog.syslog` module as to not cause confusion.

### CODE REVIEW

File: src/freenas/etc/find_alias_for_smtplib.py [hunk 1]
```diff
@@ -2,133 +2,125 @@
...
-import syslog
 
 from truenas_api_client import Client
 
-ALIASES = re.compile(r'^(?P<from>[^#]\S*?):\s*(?P<to>\S+)$')
-
 
 def do_sendmail(msg, to_addrs=None, parse_recipients=False):
+    to_addrs = ["root"] if to_addrs is None else to_addrs
+    if to_addrs is None and not parse_recipients:
```
Comment (creatorcary) [inline src/freenas/etc/find_alias_for_smtplib.py:13]: This is an impossible condition right? `to_addrs` will never be `None` here

Comment (yocalebo) [reply src/freenas/etc/find_alias_for_smtplib.py:13]: `to_addrs` is a keyword argument passed to this function so it can be `None`.

Comment (yocalebo) [reply src/freenas/etc/find_alias_for_smtplib.py:13]: oh you're right, i'll fix

**Review by creatorcary — APPROVED** [review]

**Review by mgrimesix — APPROVED** [review]
Approve with comments.

File: src/freenas/etc/find_alias_for_smtplib.py (hunk 1 above)
Comment (yocalebo) [reply src/freenas/etc/find_alias_for_smtplib.py]: I pushed a new commit. We actually don't even use the `find_alias*` scripts on SCALE so we can remove all this stuff.

File: src/freenas/etc/find_alias_for_smtplib.py [hunk 2]
```diff
@@ -2,133 +2,125 @@
...
+        raise ValueError("Do not know who to send the message to.")
```
[the remark this answers is not in the document]
Comment (yocalebo) [reply src/freenas/etc/find_alias_for_smtplib.py]: I pushed a new commit. We actually don't even use the find_alias* scripts on SCALE so we can remove all this stuff.

File: src/middlewared/middlewared/plugins/system_general/update.py [hunk 3]
```diff
@@ -1,5 +1,4 @@
 import asyncio
-import syslog
 
 import middlewared.sqlalchemy as sa
```
[the remark this answers is not in the document]
Comment (yocalebo) [reply src/middlewared/middlewared/plugins/system_general/update.py:4]: I don't think so?

**Pull request merged**